includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -1098,6 +1117,213 @@ private function sanitize_disallowed_attribute_values_in_node( DOMElement $node,
 		return $attributes_pending_removal;
 	}
 
+	/**
+	 * Ensure the width and height for the element are valid.
+	 *
```

Comment:
For all of the following methods, add a reference and link to the JS code that this was adapted from, and at which version.
As we copy behavior from the JS validator, we'll need to make sure we keep it in sync over time.